src/client/signatures.rs

```diff
@@ -86,3 +75,184 @@ impl From<CanonicalJSONError> for SignatureError {
         err.into()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{Collection, Verification};
+    use env_logger;
+    use httpmock::Method::GET;
+    use httpmock::{Mock, MockServer};
+    use serde_json::json;
+
+    fn verify_signature(
+        mock_server: &MockServer,
+        collection: Collection,
+        certificate: &str,
+        should_fail: bool,
+    ) {
+        let mut get_pem_certificate = Mock::new()
+            .expect_method(GET)
+            .expect_path(
+                "/chains/remote-settings.content-signature.mozilla.org-2020-09-04-17-16-15.chain",
+            )
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_body(certificate)
+            .create_on(&mock_server);
+
+        let mut verifiers: Vec<&dyn Verification> = Vec::new();
+
+        #[cfg(feature = "ring_verifier")]
+        verifiers.push(&super::ring_verifier::RingVerifier {});
+
+        #[cfg(feature = "rc_crypto_verifier")]
+        verifiers.push(&super::rc_crypto_verifier::RcCryptoVerifier {});
+
+        for verifier in &verifiers {
+            if should_fail {
+                assert!(verifier.verify(&collection).is_err())
+            } else {
+                match verifier.verify(&collection) {
+                    Err(err) => {
+                        println!("{:?}", err);
```

Comment:
Another option would be to simply unwrap the result

    verifier.verify(&collection).unwrap();

This will panic if the result isn't `Ok(_)`, and the error will be included in the output. (`assert!()` also just panics if it fails. It's not special in any way.)